block/submit.go

```diff
@@ -78,26 +70,27 @@ func SubmitLoopInner(
 				case n := <-bytesProduced:
 					pendingBytes.Add(uint64(n))
 					logger.Info("Added bytes produced to bytes pending submission counter.", "n", n)
-				case <-ticker.C:
 				}
 			}
-
-			types.RollappPendingSubmissionsSkewNumBytes.Set(float64(pendingBytes.Load()))
-			types.RollappPendingSubmissionsSkewNumBatches.Set(float64(pendingBytes.Load() / maxBatchBytes))
 			submitter.Nudge()
 		}
 	})
 
 	eg.Go(func() error {
-		// 'submitter': this thread actually creates and submits batches
+		// 'submitter': this thread actually creates and submits batches, and will do it on a timer if he isn't nudged by block production
 		timeLastSubmission := time.Now()
+		ticker := time.NewTicker(maxBatchTime / 10) // interval does not need to match max batch time since we keep track anyway, it's just to wakeup
```

Comment:
why `submitter` needs to `just to wakeup`?
u should move the `maxDataNotExceeded` and `lastSubmissionIsRecent` checks to `trigger`

Reply:
anyways happy to merge this because it's an improvement regardless